shell.nix

```
@@ -24,29 +25,45 @@ pkgs.mkShell {
     jq
     lvm2_dmeventd
     nixos-shell
+    niv
   ] ++ pkgs.lib.optional (builtins.getEnv "IN_NIX_SHELL" == "pure") [ docker-client ];
 
   PRE_COMMIT_ALLOW_NO_CONFIG = 1;
 
   shellHook = ''
+    unset GOARCH
+    unset GOOS
     unset GOROOT
-    export GOPATH=$(pwd)/nix/.go
-    export GOCACHE=$(pwd)/nix/.go/cache
-    export TMPDIR=$(pwd)/nix/.tmp
-    export PATH=$GOPATH/bin:$PATH
+    
+    # Temp directories should not be in project directory to avoid issues if it's mounted remotely
+    export CGO_ENABLED=0
+    export GOCACHE="$HOME/nix/.go/.cache"
```

Maybe something like this?

```suggestion
    export GOCACHE="$HOME/.cache/openebs/lvm-localpv/.go/.cache"
```

Btw, what's the issue with having this in the workspace when "mounted remotely"?

---

I am developing on a Mac, and using a Vagrant VM running Debian for development (no LVM on Mac). The project directory is mounted using a remote filesystem (vboxsf). Somehow, Go and the CI tests to not work when putting binaries and temp files in the project folder, but work fine if I e.g. put them into the home directory.

So basically this allows non-Linux developers work on the project.